javacord-core/src/main/java/org/javacord/core/entity/message/MessageImpl.java

```diff
@@ -185,19 +194,31 @@ public MessageImpl(DiscordApiImpl api, TextChannel channel, JsonNode data) {
             }
         }
 
-        if (data.has("mention_roles") && !data.get("mention_roles").isNull()) {
+        if (data.hasNonNull("mention_roles")) {
             getServer().ifPresent(server -> {
                 for (JsonNode roleMentionJson : data.get("mention_roles")) {
                     server.getRoleById(roleMentionJson.asText()).ifPresent(roleMentions::add);
                 }
             });
         }
 
-        if (data.has("activity") && !data.get("activity").isNull()) {
+        if (data.hasNonNull("activity")) {
             activity = new MessageActivityImpl(this, data.get("activity"));
         } else {
             activity = null;
         }
+
+        if (data.has("message_reference")) {
```

Comment:
Better always use `hasNonNull`. You cannot trust Discord or its docs.
```suggestion
        if (data.hasNonNull("message_reference")) {
```

Reply:
Changed it. :)